cinema/views.py

```diff
@@ -43,6 +46,26 @@
 
         return MovieSerializer
 
+    def get_queryset(self):
+        queryset = self.queryset
+        if self.action in ("list", "retrieve"):
+            queryset = queryset.prefetch_related("genres", "actors")
+
+        title = self.request.query_params.get("title")
+        genres = self.request.query_params.get("genres")
+        actors = self.request.query_params.get("actors")
+
+        if title:
+            queryset = queryset.filter(title__icontains=title)
+        if genres:
+            genres = [int(str_id) for str_id in genres.split(",")]
+            queryset = queryset.filter(genres__id__in=genres)
+        if actors:
+            actors = [int(str_id) for str_id in actors.split(",")]
```

Comment:
Similar to the genres filter, add error handling for the conversion of actor IDs to integers. This will prevent runtime errors if non-integer values are provided in the query string.